src/state.rs

```diff
@@ -137,23 +137,17 @@ impl VMState {
         let new_context = Context::new(program_code, gas_stipend);
         self.running_contexts.push(new_context);
     }
-    pub fn pop_context(&mut self) {
-        self.running_contexts.pop();
+    pub fn pop_context(&mut self) -> Context {
+        self.running_contexts.pop().unwrap()
```

Comment:
I think this is a good place to state what happened:
```suggestion
        self.running_contexts.pop().expect('Error: No running context')
```

Reply:
Sure!